src/Microsoft.DotNet.CodeFormatter.Analyzers/ExplicitVariableTypeFixer.cs

```diff
@@ -0,0 +1,114 @@
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            CancellationToken cancellationToken = context.CancellationToken;
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            SemanticModel model = await context.Document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            Diagnostic diagnostic = context.Diagnostics.First();
+            TextSpan diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            SyntaxNode typeSyntaxNode = null;
+            ILocalSymbol varSymbol = null;
+            switch (GetRuleType(diagnostic))
+            {
+                case RuleType.RuleVariableDeclaration:
+                    var variableDeclarationNode = root.
+                        FindNode(diagnosticSpan).
+                        FirstAncestorOrSelf<VariableDeclarationSyntax>();
+                    Debug.Assert(variableDeclarationNode != null);
+                    typeSyntaxNode = variableDeclarationNode.Type;
+                    // Implicit typed variables cannot have multiple declartors
+                    varSymbol = (ILocalSymbol)model.GetDeclaredSymbol(variableDeclarationNode.Variables.Single(), cancellationToken);
+                    break;
+
+                case RuleType.RuleForEachStatement:
+                    var forEachStatementNode = root.
+                        FindToken(diagnosticSpan.Start).
+                        Parent.
+                        FirstAncestorOrSelf<ForEachStatementSyntax>();
+                    Debug.Assert(forEachStatementNode != null);
+                    typeSyntaxNode = forEachStatementNode.Type;
+                    varSymbol = model.GetDeclaredSymbol(forEachStatementNode, cancellationToken);
+                    break;
+
+                case RuleType.None:
+                    break;
+            }
+            if (varSymbol != null)
+            {
+                DocumentEditor documentEditor = await DocumentEditor.CreateAsync(context.Document, cancellationToken);context.RegisterCodeFix(
+                    CodeAction.Create(
+                        Resources.ExplicitVariableTypeFixer_Title,
+                        c => ReplaceVarWithExplicitType(context.Document, 
+                                                        root, 
+                                                        typeSyntaxNode, 
+                                                        documentEditor.Generator.TypeExpression(varSymbol.Type).WithTriviaFrom(typeSyntaxNode))),
+                    diagnostic);
+            }
+        }
+
+        private Task<Document> ReplaceVarWithExplicitType(Document document, SyntaxNode root, SyntaxNode varNode, SyntaxNode explicitTypeNode)
+        {
+            return Task.FromResult(
```

Comment:
You could simply call documentEdtior.ReplaceNode and then return documentEdtior.GetchangedDocument()